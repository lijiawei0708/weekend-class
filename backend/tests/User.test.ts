import { createTestContext } from './__helpers'

const ctx = createTestContext()

describe('User', () => {
  test('users should return seeded data', async () => {
    const result = await ctx.client.send(`
      query UsersQuery {
        users {
          id
          email
          name
          avatar
        }
      }
    `)

    expect(result).toMatchSnapshot()
  })
  test('node should return seeded user', async () => {
    const result = await ctx.client.send(`
      query UserNodeQuery {
        node(id: "VXNlcjox") {
          ... on User {
            email
            name
            avatar
          }
        }
      }
    `)

    expect(result).toMatchSnapshot()
  })
  test('nodes should return seeded users', async () => {
    const result = await ctx.client.send(`
      query UserNodesQuery {
        nodes(ids: ["VXNlcjox", "VXNlcjoy"]) {
          ... on User {
            email
            name
            avatar
          }
        }
      }
    `)

    expect(result).toMatchSnapshot()
  })
})