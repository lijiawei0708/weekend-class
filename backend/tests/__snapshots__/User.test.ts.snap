// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User node should return seeded user 1`] = `
Object {
  "node": Object {
    "avatar": null,
    "email": "beatyshot@gmail.com",
    "name": "Yuki",
  },
}
`;

exports[`User nodes should return seeded users 1`] = `
Object {
  "nodes": Array [
    Object {
      "avatar": null,
      "email": "beatyshot@gmail.com",
      "name": "Yuki",
    },
    Object {
      "avatar": null,
      "email": "beatyneko@gmail.com",
      "name": "Yami",
    },
  ],
}
`;

exports[`User users should return seeded data 1`] = `
Object {
  "users": Array [
    Object {
      "avatar": null,
      "email": "beatyshot@gmail.com",
      "id": "VXNlcjox",
      "name": "Yuki",
    },
    Object {
      "avatar": null,
      "email": "beatyneko@gmail.com",
      "id": "VXNlcjoy",
      "name": "Yami",
    },
  ],
}
`;
